src/mpyl/steps/deploy/k8s/chart.py

```diff
@@ -573,6 +604,22 @@ def to_additional_routes(self) -> list[V1AlphaIngressRoute]:
     def to_middlewares(self) -> dict[str, V1AlphaMiddleware]:
         hosts: list[HostWrapper] = self.create_host_wrappers()
         traefik = self.deployment.traefik
+        middlewares = []
+
+        if traefik and traefik.middlewares:
+            middlewares = (
+                self._replace_placeholders(traefik.middlewares.get_value(self.target))
```

Comment:
Not super important: it would be good to make the structure of the code similar to the ingress-routes, to make it easier for the reader.

In the ingress routes, we first get the values as-is and only after we replace the placeholders:
```python
ingress_route_spec = (
    self.deployment.traefik.ingress_routes.get_value(self.target)
    if self.deployment.traefik and self.deployment.traefik.ingress_routes
    else None
)

…
if self.target == Target.PULL_REQUEST:
    ingress_route_spec = self._replace_placeholders(ingress_route_spec)
```

For the middlewares, we do it all in one go:
```python
traefik = self.deployment.traefik
if traefik and traefik.middlewares:
    middlewares = (
        self._replace_placeholders(traefik.middlewares.get_value(self.target))
        if self.target == Target.PULL_REQUEST
        else traefik.middlewares.get_value(self.target)
    )
```

These two blocks do exactly the same thing, but in different ways. So it took me a bit to figure out if there was an actual difference in logic, or just in style.

Reply:
It's because there actually is a difference in logic, hehe. The Middlewares one has to be done -before- the new structure gets made, since we use a value as a key in the new structure and the replacement code would become a lot more complicated if we also have to include key replacement.